auth_session_timeout/models/res_users.py

```
@@ -0,0 +1,103 @@
+        Defaults to current time minus delay using delay stored as config
+        param.
+        """
+        params = self.env["ir.config_parameter"]
+        delay = params._auth_timeout_get_parameter_delay()
+        if delay <= 0:
+            return False
+        return time() - delay
+
+    @api.model
+    def _auth_timeout_session_terminate(self, session):
+        """Pluggable method for terminating a timed-out session
+
+        This is a late stage where a session timeout can be aborted.
+        Useful if you want to do some heavy checking, as it won't be
+        called unless the session inactivity deadline has been reached.
+
+        Return:
+            True: session terminated
+            False: session timeout cancelled
+        """
+        if session.db and session.uid:
+            session.logout(keep_db=True)
+        return True
+
+    @api.model
+    def _auth_timeout_check(self):
+        """Perform session timeout validation and expire if needed."""
+
+        if not http.request:
+            return
+
+        session = http.request.session
+
+        # Calculate deadline
+        deadline = self._auth_timeout_deadline_calculate()
+
+        # Check if past deadline
+        expired = False
+        if deadline is not False:
+            path = http.root.session_store.get_session_filename(session.sid)
+            try:
+
+                expired = getmtime(path) < deadline
+            except OSError:
+                _logger.exception(
+                    "Exception reading session file modified time.",
+                )
+                # Force expire the session. Will be resolved with new session.
+                expired = True
+
+        # Try to terminate the session
+        terminated = False
+        if expired:
+            terminated = self._auth_timeout_session_terminate(session)
+
+        # If session terminated, all done
+        if terminated:
+            raise SessionExpiredException("Session expired")
```

Hi @bosd , 
> I can replicate this behaviour..
> Any ideas why it is not working?

I see this line making the issue
Because you raise an exception here, then everything is reverted, the user can not log out

For solution, we should `return` here, not `raise`

---

@bosd please help to check, thank you

---

> For solution, we should return here, not raise

I'm having some personal time. Later, in a couple of weeks I will get back to this.

---

> For solution, we should `return` here, not `raise`

@Leonidas-VII  Thanks for the suggestion, I made the commit. It does not seem to resolve the problem.

I have the impression it is related to these lines:

https://github.com/OCA/server-auth/pull/507/files#diff-1648a6bceb7d2da7ef9fc6326f0f894ee39f11f613d80c83ae53849fdb5698eaR51-R52

---

For `return`, we can just return anything here, nothing is done with the result. So we don't need to return the exception.

Meanwhile, the issue of a refresh not leading to an expiration I think is caused by [this](https://github.com/OCA/server-auth/pull/535) - can you cherry-pick it in?

---

just a question: why don't we add the auto-reload function after 5/10 seconds instead of clicking on the button "OK"?